ibis/pyspark/compiler.py

```diff
@@ -1052,76 +1055,88 @@ def compile_window_op(t, expr, scope, timecontext, **kwargs):
     ]
 
     order_by = window._order_by
+    # Timestamp needs to be cast to long for filtering in spark
     ordering_keys = [
-        key.to_expr().get_name()
-        for key in map(operator.methodcaller('op'), order_by)
+        F.col(expr.get_name()).cast('long')
+        if isinstance(expr.op().args[0], types.TimestampColumn)
+        else expr.get_name()
+        for expr in order_by
     ]
-
     context = AggregationContext.WINDOW
     pyspark_window = Window.partitionBy(grouping_keys).orderBy(ordering_keys)
 
     # If the operand is a shift op (e.g. lead, lag), Spark will set the window
     # bounds. Only set window bounds here if not a shift operation.
     if not isinstance(operand.op(), ops.ShiftBase):
-        start = (
-            -window.preceding
-            if window.preceding is not None
-            else Window.unboundedPreceding
-        )
-        end = (
-            window.following
-            if window.following is not None
-            else Window.unboundedFollowing
-        )
-        pyspark_window = pyspark_window.rowsBetween(start, end)
+        if window.preceding is None:
+            start = Window.unboundedPreceding
+        else:
+            start = -_canonicalize_interval(window.preceding)
+        if window.following is None:
+            end = Window.unboundedFollowing
+        else:
+            end = _canonicalize_interval(window.following)
 
-    result = t.translate(
-        operand, scope, timecontext, window=pyspark_window, context=context
+        if isinstance(window.preceding, ir.IntervalScalar) or isinstance(
+            window.following, ir.IntervalScalar
+        ):
+            pyspark_window = pyspark_window.rangeBetween(start, end)
+        else:
+            pyspark_window = pyspark_window.rowsBetween(start, end)
+
+    result = t.translate(operand, scope, timecontext, context=context).over(
+        pyspark_window
     )
+
+    # These ops need type cast after running over window
```

Comment:
As we agreed `over.(window)` should be lifted to this function instead of passing via `kwargs`.
This for sure works for `compile_aggregator`, but there are some other `op`s that is dispatched in this line. For example, `lead`, `lag`, `row_numbers`, `first`, `last`, etc. We used to pass `window` as a param to them.
I removed `window` param in these functions, but there are some ops like the three ones I listed here, they do type cast after running over window. Let me know if this change makes sense to you, or maybe there are better ways to do so? @icexelloss

Reply:
I see. This is probably why `over(window)` was not in `compile_window_op` originally. 

I don't think what you have here is much better than original implementation actually and I think we should roll back to the original implementation.

Reply:
I agree branching too much isn't a good idea, I reverted and commented below for thoughts